features/netsocket/SocketAddress.cpp

```diff
@@ -122,6 +122,7 @@ static void ipv6_from_address(uint8_t *bytes, const char *addr)
     // Move suffix to end
     memmove(&shorts[NSAPI_IPv6_BYTES/2-suffix], &shorts[0],
             suffix*sizeof(uint16_t));
+    memset(&shorts[0], 0, suffix*sizeof(uint16_t));
```

Comment:
The reason I mention adding an assert is because you have no array bounds check for shorts and you get the size to zero from a separate function. Clobbered stacks can be tricky to track down, since a crash typically occurs on the function return.